glommio/src/sys/uring.rs

```diff
@@ -1736,24 +1671,30 @@ impl Reactor {
         }
     }
 
-    /// We want to go to sleep, but we can only go to sleep in one of the rings,
-    /// as we only have one thread. There are more than one sleepable rings, so
-    /// what we do is we take advantage of the fact that the ring's `ring_fd` is
-    /// pollable and register a `POLL_ADD` event into the ring we will wait on.
-    ///
-    /// We may not be able to register an `SQE` at this point, so we return an
-    /// Error and will just not sleep.
-    fn link_rings_and_sleep(&self, ring: &mut SleepableRing) -> io::Result<()> {
-        let mut link_rings = Source::new(
-            IoRequirements::default(),
-            self.link_fd,
-            SourceType::LinkRings,
-            None,
-            None,
-        );
-        ring.sleep(&mut link_rings)
-            .or_else(Self::busy_ok)
-            .map(|_| {})
+    /// Sleep until we get an event or for the specified duration (whichever
+    /// comes first)
+    fn sleep(&self, timeout: Option<Duration>) -> io::Result<()> {
+        defer!({
+            // Flush the eventfd!
+            let _ = read_eventfd(self.cqe_events_fd);
```

Comment:
Is this the correct thing to do if the poll call returned because of an event from the remote notifier?